core/modules/locale/locale.install

```diff
@@ -219,15 +219,8 @@ function locale_update_dependencies() {
  * for the data elsewhere.
  */
 function locale_update_1000() {
-  $subquery = db_select('locales_source', 'ls')
```

Comment:
Maybe a weird idea, but... could we use a switch based on `module_exists('i18n')` in this hook? Not sure.

Reply:
Do you mean that if i18n is *not* present that we delete these all? That's possible.